### PR TITLE
Kokkos: addding a section about the kokkos programming model

### PR DESCRIPTION
**Pull request by lucbv**
The section is adding in the data services general section. It currently focuses on the technical details of the Kokkos programming model itself. An additional section on the motivation for the Kokkos ecosystem, its impact on other Trilinos packages  and other aspects such as the HPSF can be added in the introduction as suggested by Michael.

### CODE REVIEW

**Comment by lucbv** [comment]
@crtrott please feel free to modify/re-write this section, I intend it to be more technical in nature and suggest that another section be added to the introduction as mentioned by Michael. In that introduction section it would be good to explain why Kokkos was create, how it impacted other packages and where it is going with HPSF, c++, etc...

**Comment by crtrott** [comment]
I will work on this today and tomorrow, before merging this please :-) (I will work on Lucs repo).

**Comment by mayrmt** [comment]
@crtrott, did you have a chance to look into the Kokkos section for the Trilinos paper?

**Comment by cgcgcg** [comment]
@crtrott Can we merge this, or did you want to make changes?

**Comment by crtrott** [comment]
I have an update ready to push. Talking over it with Luc right now.

File: introduction.tex [hunk 1]
```diff
@@ -16,7 +16,32 @@ \subsection{Modern Trilinos}
...
+in the Top500 use GPUs, with only a single system being CPUs only.
+A complication in this shift is the diversification of vendor provided programming models.
+The aforementioned 10 system actually have four different preferred programming models:
+CUDA, HIP and OneAPI SYCL for the GPU based systems and OpenMP for the CPU system.
+
+To avoid the necessity of diverging code paths Trilinos is leveraging the Kokkos Ecosystem~\cite{trott2021kokkos}
+to write harware agnostic code. The Kokkos Ecosystem started as native Trilinos packages
+Kokkos and Kokkos Kernels but split off a decade ago into a standalone project hosted at \url{https://github.com/kokkos}.
+However, Trilinos provides snapshots of the two primary Kokkos subprojects which reflect
+the latest release of Kokkos. Trilinos can also be build against version-compatible
```
Comment (lucbv) [inline introduction.tex:34]:
```suggestion
the latest release of Kokkos. Trilinos can also be built against version-compatible
```

**Review by lucbv — COMMENTED** [review]
Looks good to me, the small change for build vs built is actually more of a preference, apparently both are fine...

File: introduction.tex (hunk 1 above)
Comment (mayrmt) [reply introduction.tex]: Done.

**Comment by mayrmt** [comment]
@lucbv @crtrott, thanks for filling in some blanks in the manuscript. I'd be happy to merge the current state. Ok with you?

**Comment by crtrott** [comment]
I think so.

**Pull request merged**